src/rgw/rgw_common.h

```diff
@@ -1251,6 +1253,12 @@ struct req_state {
 
   RGWUserInfo *user;
 
+  /* Object having the knowledge about an authenticated identity and allowing
+   * to apply it during the authorization phase (verify_permission() methods
+   * of a given RGWOp). Thus, it bounds authentication and authorization steps
+   * through a well-defined interface. For more details, see rgw_auth.h. */
+  std::unique_ptr<RGWIdentityApplier> auth_identity;
```

Comment:
During the authentication phase (`RGWHandler_REST::authorize`). As a fallback `rgw_auth_transform_old_authinfo` also may allocate that.